hs_data_services/hs_data_services_sync/utilities.py

```diff
@@ -27,23 +32,36 @@ def update_data_services(resource_id):
 
         for db in database_list['geoserver']['unregister']:
             unregister_geoserver_db(resource_id, db)
+            remove_copied_file_from_geoserver(resource_id, db)
 
         for db in database_list['geoserver']['register']:
+            # copy geoserver files from HS to GeoServer
+            file_transfer_info = copy_file_to_geoserver(resource_id, db)
+            if file_transfer_info['success'] is False:
+                response['message'] = file_transfer_info['message']
+                return response
             db_info = register_geoserver_db(resource_id, db)
             if db_info['success'] is False:
                 unregister_geoserver_db(resource_id, db)
+                # TODO: ideally we "inform" HS that the registration failed
+                # This is called from an async task, so we can't return a meaningful response to HS
```

Comment:
for now, Martin suggested that I just update the 404 message from Geoserver to say something like "we are still creating your layers, check back in a few minutes. Contact us if this issue doesn't resolve"